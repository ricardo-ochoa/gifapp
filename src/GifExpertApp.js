import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

import { gsap } from "gsap";


// import PropTypes from 'prop-types';

const GifExpertApp = ({ defaultCategories = ['Luffy']}) => {

    // const [categories, setCategories] = useState(['One Piece']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = (e) => {
    //     // Dos formas de aregar elementos en un arreglo en un componente:
    //     // setCategories([...categories, 'HunterxHunter']); 
    //     setCategories( cats => [...cats, 'HunterxHunter']); 
    // }

    let timeline = gsap.timeline();

    return (
        <>
            <h2> GifExpertApp </h2>
            <AddCategory setCategories = { setCategories }/>
            <hr></hr>

                {
                    categories.map( category => (
                        <ol timeline = {timeline} id='container'>
                        <GifGrid 
                            key = {category}
                            category = {category}
                        />
                        </ol>
                    ))
                }
        </>
    );
}

GifExpertApp.propTypes = {
    // value: PropTypes.number
}

export default GifExpertApp;