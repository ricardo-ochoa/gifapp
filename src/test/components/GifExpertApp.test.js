import React from "react";
import GifExpertApp from "../../GifExpertApp";
import { shallow } from 'enzyme'

describe('Prueba de GifExpertApp', () => {

    test('Hacer match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />) 
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar una lista de categorÃ­as', () => {

        const categories = ['Naruto', 'HxH'];
        const wrapper = shallow( <GifExpertApp defaultCategories = { categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
}) 