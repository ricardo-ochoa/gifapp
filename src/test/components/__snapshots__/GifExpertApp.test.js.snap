// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol
    id="container"
    timeline={
      Timeline {
        "_delay": 0,
        "_dp": Timeline {
          "_delay": 0,
          "_dirty": 1,
          "_dur": 0,
          "_first": Timeline {
            "_delay": 0,
            "_dp": [Circular],
            "_dur": 0,
            "_end": 0,
            "_next": [Circular],
            "_prev": undefined,
            "_repeat": 0,
            "_sort": true,
            "_start": 0,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": false,
            "data": undefined,
            "labels": Object {},
            "parent": [Circular],
            "smoothChildTiming": false,
            "vars": Object {},
          },
          "_last": [Circular],
          "_recent": [Circular],
          "_repeat": 0,
          "_sort": false,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": true,
          "data": undefined,
          "labels": Object {},
          "smoothChildTiming": true,
          "vars": Object {
            "autoRemoveChildren": true,
            "defaults": Object {
              "delay": 0,
              "duration": 0.5,
              "ease": [Function],
              "overwrite": false,
            },
            "id": "root",
            "smoothChildTiming": true,
            "sortChildren": false,
          },
        },
        "_dur": 0,
        "_end": 0,
        "_next": undefined,
        "_prev": Timeline {
          "_delay": 0,
          "_dp": Timeline {
            "_delay": 0,
            "_dirty": 1,
            "_dur": 0,
            "_first": [Circular],
            "_last": [Circular],
            "_recent": [Circular],
            "_repeat": 0,
            "_sort": false,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": true,
            "data": undefined,
            "labels": Object {},
            "smoothChildTiming": true,
            "vars": Object {
              "autoRemoveChildren": true,
              "defaults": Object {
                "delay": 0,
                "duration": 0.5,
                "ease": [Function],
                "overwrite": false,
              },
              "id": "root",
              "smoothChildTiming": true,
              "sortChildren": false,
            },
          },
          "_dur": 0,
          "_end": 0,
          "_next": [Circular],
          "_prev": undefined,
          "_repeat": 0,
          "_sort": true,
          "_start": 0,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": false,
          "data": undefined,
          "labels": Object {},
          "parent": Timeline {
            "_delay": 0,
            "_dirty": 1,
            "_dur": 0,
            "_first": [Circular],
            "_last": [Circular],
            "_recent": [Circular],
            "_repeat": 0,
            "_sort": false,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": true,
            "data": undefined,
            "labels": Object {},
            "smoothChildTiming": true,
            "vars": Object {
              "autoRemoveChildren": true,
              "defaults": Object {
                "delay": 0,
                "duration": 0.5,
                "ease": [Function],
                "overwrite": false,
              },
              "id": "root",
              "smoothChildTiming": true,
              "sortChildren": false,
            },
          },
          "smoothChildTiming": false,
          "vars": Object {},
        },
        "_repeat": 0,
        "_sort": true,
        "_start": 0,
        "_tDur": 0,
        "_ts": 1,
        "autoRemoveChildren": false,
        "data": undefined,
        "labels": Object {},
        "parent": Timeline {
          "_delay": 0,
          "_dirty": 1,
          "_dur": 0,
          "_first": Timeline {
            "_delay": 0,
            "_dp": [Circular],
            "_dur": 0,
            "_end": 0,
            "_next": [Circular],
            "_prev": undefined,
            "_repeat": 0,
            "_sort": true,
            "_start": 0,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": false,
            "data": undefined,
            "labels": Object {},
            "parent": [Circular],
            "smoothChildTiming": false,
            "vars": Object {},
          },
          "_last": [Circular],
          "_recent": [Circular],
          "_repeat": 0,
          "_sort": false,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": true,
          "data": undefined,
          "labels": Object {},
          "smoothChildTiming": true,
          "vars": Object {
            "autoRemoveChildren": true,
            "defaults": Object {
              "delay": 0,
              "duration": 0.5,
              "ease": [Function],
              "overwrite": false,
            },
            "id": "root",
            "smoothChildTiming": true,
            "sortChildren": false,
          },
        },
        "smoothChildTiming": false,
        "vars": Object {},
      }
    }
  >
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
  <ol
    id="container"
    timeline={
      Timeline {
        "_delay": 0,
        "_dp": Timeline {
          "_delay": 0,
          "_dirty": 1,
          "_dur": 0,
          "_first": Timeline {
            "_delay": 0,
            "_dp": [Circular],
            "_dur": 0,
            "_end": 0,
            "_next": [Circular],
            "_prev": undefined,
            "_repeat": 0,
            "_sort": true,
            "_start": 0,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": false,
            "data": undefined,
            "labels": Object {},
            "parent": [Circular],
            "smoothChildTiming": false,
            "vars": Object {},
          },
          "_last": [Circular],
          "_recent": [Circular],
          "_repeat": 0,
          "_sort": false,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": true,
          "data": undefined,
          "labels": Object {},
          "smoothChildTiming": true,
          "vars": Object {
            "autoRemoveChildren": true,
            "defaults": Object {
              "delay": 0,
              "duration": 0.5,
              "ease": [Function],
              "overwrite": false,
            },
            "id": "root",
            "smoothChildTiming": true,
            "sortChildren": false,
          },
        },
        "_dur": 0,
        "_end": 0,
        "_next": undefined,
        "_prev": Timeline {
          "_delay": 0,
          "_dp": Timeline {
            "_delay": 0,
            "_dirty": 1,
            "_dur": 0,
            "_first": [Circular],
            "_last": [Circular],
            "_recent": [Circular],
            "_repeat": 0,
            "_sort": false,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": true,
            "data": undefined,
            "labels": Object {},
            "smoothChildTiming": true,
            "vars": Object {
              "autoRemoveChildren": true,
              "defaults": Object {
                "delay": 0,
                "duration": 0.5,
                "ease": [Function],
                "overwrite": false,
              },
              "id": "root",
              "smoothChildTiming": true,
              "sortChildren": false,
            },
          },
          "_dur": 0,
          "_end": 0,
          "_next": [Circular],
          "_prev": undefined,
          "_repeat": 0,
          "_sort": true,
          "_start": 0,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": false,
          "data": undefined,
          "labels": Object {},
          "parent": Timeline {
            "_delay": 0,
            "_dirty": 1,
            "_dur": 0,
            "_first": [Circular],
            "_last": [Circular],
            "_recent": [Circular],
            "_repeat": 0,
            "_sort": false,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": true,
            "data": undefined,
            "labels": Object {},
            "smoothChildTiming": true,
            "vars": Object {
              "autoRemoveChildren": true,
              "defaults": Object {
                "delay": 0,
                "duration": 0.5,
                "ease": [Function],
                "overwrite": false,
              },
              "id": "root",
              "smoothChildTiming": true,
              "sortChildren": false,
            },
          },
          "smoothChildTiming": false,
          "vars": Object {},
        },
        "_repeat": 0,
        "_sort": true,
        "_start": 0,
        "_tDur": 0,
        "_ts": 1,
        "autoRemoveChildren": false,
        "data": undefined,
        "labels": Object {},
        "parent": Timeline {
          "_delay": 0,
          "_dirty": 1,
          "_dur": 0,
          "_first": Timeline {
            "_delay": 0,
            "_dp": [Circular],
            "_dur": 0,
            "_end": 0,
            "_next": [Circular],
            "_prev": undefined,
            "_repeat": 0,
            "_sort": true,
            "_start": 0,
            "_tDur": 0,
            "_ts": 1,
            "autoRemoveChildren": false,
            "data": undefined,
            "labels": Object {},
            "parent": [Circular],
            "smoothChildTiming": false,
            "vars": Object {},
          },
          "_last": [Circular],
          "_recent": [Circular],
          "_repeat": 0,
          "_sort": false,
          "_tDur": 0,
          "_ts": 1,
          "autoRemoveChildren": true,
          "data": undefined,
          "labels": Object {},
          "smoothChildTiming": true,
          "vars": Object {
            "autoRemoveChildren": true,
            "defaults": Object {
              "delay": 0,
              "duration": 0.5,
              "ease": [Function],
              "overwrite": false,
            },
            "id": "root",
            "smoothChildTiming": true,
            "sortChildren": false,
          },
        },
        "smoothChildTiming": false,
        "vars": Object {},
      }
    }
  >
    <GifGrid
      category="HxH"
      key="HxH"
    />
  </ol>
</Fragment>
`;

exports[`Prueba de GifExpertApp Hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
